keras_nlp/src/models/clip/clip_text_encoder.py

```diff
@@ -63,64 +60,45 @@ def __init__(
             for _ in range(num_layers)
         ]
         self.layer_norm = layers.LayerNormalization(
-            epsilon=0.00001, dtype=dtype, name="layer_norm"
-        )
-        self.text_projection = layers.Dense(
-            hidden_dim,
-            use_bias=False,
-            dtype=dtype,
-            name="text_projection",
+            epsilon=1e-6, dtype="float32", name="layer_norm"
         )
 
         # === Functional Model ===
-        encoder_token_ids = layers.Input(
-            shape=(sequence_length,), dtype="int32", name="encoder_token_ids"
+        token_id_input = layers.Input(
+            shape=(max_sequence_length,), dtype="int32", name="token_ids"
```

Comment:
Can we keep this shape `(None,)` so you don't have to pass the max_sequence_length?

Reply:
Unfortunately, no.
`max_sequence_length` is also required by `TokenAndPositionEmbedding` and it cannot be set to `None`.

Reply:
Oh sorry, I was wondering if we could have `max_sequence_length` as a parameter, pass it to the embedding, but keep the `keras.Input` shape as `(None,)`. This would allow calling the model with sequence lengths shorter than the max. That's what we do for other models using `TokenAndPositionEmbedding` in this repo.

Reply:
Got it. Thanks for the clarification. Fixed.